### ISSUE TITLE
RFC: Prioritise Platforms to Support

### ISSUE THREAD

**Issue by IGBC** [Rust-Retro/RWG#1]
A "non exhaustive" list of platforms was provided by @thejpster, shown below that currently do not have mainstream rust support. All of these plus countless more are candidates for RWG support:

    Windows 9x
    Windows NT 4.0 for x86, Alpha and PowerPC
    68k Classic Mac (up to System 7)
    PPC Classic Mac (Mac OS 8 and 9)
    PPC Mac OS X (Mac OS X 10.0 to 10.4)
    68k Amiga
    PPC Amiga
    68k Atari ST
    XBOX and Xbox 360
    PlayStation 1, 2 and 3
    Wii, GameCube, N64, GBA
    ARM v2 Acorn Archimedes
    ARM v3 Acorn RISC PC
    Sun UltraSPARC
    6502 machines (e.g. C64)
    6809 machines (e.g. TI TRS-80)
    Z80 machines (e.g. ZX Spectrum)

This RFC requests input on which platforms we should prioritise (on this list or not) and how we should choose priorities in the future.

**Comment by spacekatia** [comment]
The original gameboy is a z80, isn't it?

**Comment by IGBC** [comment]
> The original gameboy is a z80, isn't it?

Yes it is. 

While we're at it the GBA's are ARM v7

**Comment by mcdoll** [comment]
> > The original gameboy is a z80, isn't it?
> 
> Yes it is.
> 
> While we're at it the GBA's are ARM v7

No, it is an ARM7, which makes it ARMv4 (the arm devs were really smart about their numbering scheme). ARMv4 seems to have rust support for linux (tier 3), so baremetal should be possible, too.

**Comment by thejpster** [comment]
> > > The original gameboy is a z80, isn't it?
> > 
> > 
> > Yes it is.
> > While we're at it the GBA's are ARM v7
> 
> No, it is an ARM7, which makes it ARMv4 (the arm devs were really smart about their numbering scheme). ARMv4 seems to have rust support for linux (tier 3), so baremetal should be possible, too.

I vaguely recall someone doing Rust on a GBA already. It came up when I was looking at ARMv6 support to run Rust on a Raspberry Pi Zero / 1.

While we're here:

6502 - Every Commodore 8 bit machine, Apple I, Apple II, NES, Acorn 8 bit range (including BBC Micro), Bender from Futurama, and the Cyberdyne Systems Model 101 Series 800
6809 - TRS CoCo, Dragon 32
Z80 - Every CP/M machine, entire Amstrad 8 bit range, entire Sinclair 8 bit range
PowerPC 'G3' - GameCube and Wii
PowerPC 'G5' - Xbox 360 (+ PS3?)
MIPS R5900 - PS2

**Comment by mcdoll** [comment]
Cool, I didn't know about the gba support for rust. There seems to be already a small community doing rust for gaming consoles: https://github.com/rust-console

(concerning pi-zero: I started programming that, but I switched to the beaglebone, since the documentation for the raspis is horrible; but I guess armv6/v7 is too modern for rust-retro)

**Comment by IGBC** [comment]
> Cool, I didn't know about the gba support for rust. There seems to be already a small community doing rust for gaming consoles: https://github.com/rust-console

One to go on the associated projects list:

> (concerning pi-zero: I started programming that, but I switched to the beaglebone, since the documentation for the raspis is horrible; but I guess armv6/v7 is too modern for rust-retro)

Yes for modern arm check out the embedded working group. Beaglebone and RPi's are also supported in mainline when linux is installed.

**Comment by IGBC** [comment]
There's old MIPS

Old UNIX workstations etc

**Comment by thejpster** [comment]
Ah, the SGI Indy. MIPS R4000, running IRIX.

**Comment by IGBC** [comment]
I own an indigo 2 with an R8000

**Comment by sajattack** [comment]
> I vaguely recall someone doing Rust on GBA already.

Paging @lokathor

**Comment by Lokathor** [comment]
Uhm, hi? I already have a `rust-console` group and would rather that we join efforts than form a whole new group.

PS: I haven't read the topic until now i just got the email suddenly

**Comment by IGBC** [comment]
@Lokathor log into our matrix room and we can talk about this face to face. 

An internal guthub issue isbt the place for politics

Our matrix chat is linked in the readme of this repo

**Comment by spacekatia** [comment]
IBM AIX
https://en.m.wikipedia.org/wiki/IBM_AIX

**Comment by ponyatov** [comment]
I found some PC-controlled hardware in use that uses WinNT & Windows 2000 (industrial control, and CNC)
